### PR TITLE
Adding migrations support.

### PR DESCRIPTION
**Pull request by youtous**
Like others plugins, we should use the Migrations plugin.

> It allows you to evolve your database tables over time. Instead of writing schema modifications in SQL, this plugin allows you to use an intuitive set of methods to implement your database changes.

### CODE REVIEW

**Comment by tomcur** [comment]
Thank you! This is a great improvement.

Could you change `series` to be before `token` in the column ordering? Also change `series` to have a unique index (primary key). The `token` column should not be unique. It is the `series` column that identifies a user. The `token` is like a password.

**Comment by youtous** [comment]
Indeed, I inversed token and series. You want the `series` be a composed pk ? [`id,series`] or just adding an unique constraint onto the `series`?
I guess you talked about constraint and not pk.

**Comment by youtous** [comment]
I added a .gitignore, composer comes with his fat vendor folder, feel free to edit for your needs 💃

**Comment by tomcur** [comment]
I used to only have `id` be unique, but actually it's better to have the `series` column be unique. It does not need to be a composed primary key, as `id` and `series` are both unique on their own.

An `id` is not strictly necessary, but if we don't have an `id` as primary key I think we need to change `AuthTokensTable` to point to `series` as its primary key; but this will introduce some issues with backwards compatibility. E.g. CakePHP >= 3.4 uses `$this->setPrimaryKey($key)`, whereas CakePHP < 3.4 uses `$this->primaryKey($key)`.

In short, I think we should add `id` as a primary key, and place a unique index on `series`.

**Comment by youtous** [comment]
Ok! We should think to do some CI test for avoid backwards compatibility 🤔 

Integer id are faster than string, so we should keep the id for CRUD operations.
I have the idea to allow the User to revoke particular tokens (CookieTokens), for this kind of action, integer id are nice :)

The `AbstractMigration` comes with `autoid `activated by default.

**Comment by tomcur** [comment]
Ah, great, so an `id` primary key is added already. 👍 I'll merge the PR.

Some way to (semi-)automatically test changes would be nice. Do you have suggestions on how to accomplish that?

**Comment by youtous** [comment]
Using [Travis CI](https://docs.travis-ci.com/user/getting-started/) this should not be complicated.

We can do something like it is explained [here](https://dyrynda.com.au/blog/testing-multiple-dependency-versions-with-travis-ci)

Anyway, starting to use **phpunit** and **Travis** is the good way to start.

If you want to see how other plugins do it, take a tour [here](https://github.com/dereuromark/cakephp-queue/blob/master/.travis.yml).

The longer part will be to write tests for the existing codebase 😹
